api/rolestore/client.go

```diff
@@ -19,734 +20,732 @@ type RoleStore struct {
+// GetUsersAuthorizedKeys get users authorized keys.
+func (c *RoleStore) GetUsersAuthorizedKeys(userID string, opts ...filters.Option) (*response.ResultSet[AuthorizedKey], error) {
+	keys := &response.ResultSet[AuthorizedKey]{}
+	params := url.Values{}
+
+	for _, opt := range opts {
+		opt(&params)
 	}
 
-	_, err := store.api.
-		URL("/role-store/api/v1/roles/%s/principalkeys/generate", url.PathEscape(roleID)).
-		Post(nil, &object)
+	_, err := c.api.
+		URL("/role-store/api/v1/users/%s/authorizedkeys", userID).
+		Query(params).
+		Get(&keys)
 
-	return object.ID, err
+	return keys, err
 }
 
-// ImportPrincipalKey import new principal key for existing role
-func (store *RoleStore) ImportPrincipalKey(key PrivateKey, roleID string) (string, error) {
-	var object struct {
-		ID string `json:"id"`
-	}
+// CreateAuthorizedKey create authorized key for user.
+func (c *RoleStore) CreateAuthorizedKey(userID string, key *AuthorizedKey) (response.Identifier, error) {
+	identifier := response.Identifier{}
 
-	_, err := store.api.
-		URL("/role-store/api/v1/roles/%s/principalkeys/import", url.PathEscape(roleID)).
-		Post(&key, &object)
+	_, err := c.api.
+		URL("/role-store/api/v1/users/%s/authorizedkeys", userID).
+		Post(&key, &identifier)
 
-	return object.ID, err
+	return identifier, err
 }
 
-// PrincipalKey returns a role's principal key object.
-func (store *RoleStore) PrincipalKey(roleID, keyID string) (*PrincipalKey, error) {
-	key := &PrincipalKey{}
+// GetUsersAuthorizedKey get users authorized key by id.
+func (c *RoleStore) GetUsersAuthorizedKey(userID, keyID string) (*AuthorizedKey, error) {
+	key := &AuthorizedKey{}
 
-	_, err := store.api.
-		URL("/role-store/api/v1/roles/%s/principalkeys/%s", url.PathEscape(roleID), url.PathEscape(keyID)).
-		Get(key)
+	_, err := c.api.
+		URL("/role-store/api/v1/users/%s/authorizedkeys/%s", userID, keyID).
+		Get(&key)
 
 	return key, err
 }
 
-// DeletePrincipalKey delete a role's principal key
-func (store *RoleStore) DeletePrincipalKey(roleID, keyID string) error {
-	_, err := store.api.
-		URL("/role-store/api/v1/roles/%s/principalkeys/%s", roleID, keyID).
+// UpdateUsersAuthorizedKey update users authorized key.
+func (c *RoleStore) UpdateUsersAuthorizedKey(userID, keyID string, key *AuthorizedKey) error {
+	_, err := c.api.
+		URL("/role-store/api/v1/users/%s/authorizedkeys/%s", userID, keyID).
+		Put(&key)
+
+	return err
+}
+
+// DeleteUsersAuthorizedKey delete a users authorized key.
+func (c *RoleStore) DeleteUsersAuthorizedKey(userID, keyID string) error {
+	_, err := c.api.
+		URL("/role-store/api/v1/users/%s/authorizedkeys/%s", userID, keyID).
 		Delete()
 
 	return err
 }
 
-// User gets information about the argument user ID.
-func (store *RoleStore) User(userID string) (*User, error) {
-	user := &User{}
+// GetCurrentUsersAuthorizedKeys get current users authorized keys.
+func (c *RoleStore) GetCurrentUsersAuthorizedKeys(opts ...filters.Option) (*response.ResultSet[AuthorizedKey], error) {
```

Comment:
For all the functions below with /current/ in path: decide which pattern to name:
- <Get>CurrentUser<Stuff>, or
- <Get><Stuff>CurrentUser

Apply the same pattern to all eligible functions after this line.